sdk/src/sharedTest/java/io/customer/sdk/queue/QueueStorageIntegrationTest.kt

```diff
@@ -109,6 +109,49 @@ class QueueStorageIntegrationTest : BaseTest() {
         queueStorage.getInventory().count() shouldBeEqualTo 0
     }
 
+    @Test
+    fun givenDeleteGroupTask_expectAllTasksInGroupToDelete() {
+        val givenStartOfTheGroup = QueueTaskGroup.IdentifyProfile(String.random)
+
+        queueStorage.create(String.random, String.random, givenStartOfTheGroup, null)
+        queueStorage.create(String.random, String.random, null, listOf(givenStartOfTheGroup))
+        queueStorage.create(String.random, String.random, null, listOf(givenStartOfTheGroup))
```

Comment:
What about this edge case: 

What if we had this:
```
```
```suggestion
        queueStorage.create(String.random, String.random, groupStartOfAnotherGroup, listOf(givenStartOfTheGroup))
        queueStorage.create(String.random, String.random, null, listOf(givenStartOfTheGroup))
```

So a task was the member of a group *and* was also the start of a separate group. 

If *that* task gets deleted (because it's a member of a group), should we also delete all of the tasks that *it* starts? 

I think that register push token is one of these edge cases. It belongs to the identify profile group but it also starts it's own group. Delete device token is a member of that group that register starts.